src/IO/ADIOS/ADIOS2IOHandler.cpp

```diff
@@ -20,58 +20,1244 @@
+    AbstractIOHandler * handler,
+    MPI_Comm communicator )
+    : AbstractIOHandlerImplCommon( handler )
+    , m_comm{ communicator }
+    , m_ADIOS{ communicator, ADIOS2_DEBUG_MODE }
 {
-    return m_impl->flush();
 }
 
-#endif
+#endif // openPMD_HAVE_MPI
 
-#if openPMD_HAVE_ADIOS2 && !openPMD_HAVE_MPI
-ADIOS2IOHandler::ADIOS2IOHandler(std::string path, AccessType at)
-        : AbstractIOHandler(std::move(path), at),
-          m_impl{new ADIOS2IOHandlerImpl(this)}
-{ }
+ADIOS2IOHandlerImpl::ADIOS2IOHandlerImpl( AbstractIOHandler * handler )
+    : AbstractIOHandlerImplCommon( handler ), m_ADIOS{ ADIOS2_DEBUG_MODE }
+{
+}
 
-ADIOS2IOHandler::~ADIOS2IOHandler() = default;
+ADIOS2IOHandlerImpl::~ADIOS2IOHandlerImpl( )
+{
+    this->flush( );
+}
 
-std::future< void >
-ADIOS2IOHandler::flush()
+std::future< void > ADIOS2IOHandlerImpl::flush( )
 {
-    return m_impl->flush();
+    auto res = AbstractIOHandlerImpl::flush( );
+    for ( auto & p : m_fileData )
+    {
+        if ( m_dirty.find( p.first ) != m_dirty.end( ) )
+        {
+            p.second->flush( );
+        }
+        else
+        {
+            p.second->drop( );
+        }
+    }
+    return res;
 }
-#else
-ADIOS2IOHandler::ADIOS2IOHandler(std::string path, AccessType at)
+
+void ADIOS2IOHandlerImpl::createFile(
+    Writable * writable,
+    Parameter< Operation::CREATE_FILE > const & parameters )
+{
+    VERIFY_ALWAYS( m_handler->accessTypeBackend != AccessType::READ_ONLY,
+                   "Creating a file in read-only mode is not possible." );
+
+    if ( !writable->written )
+    {
+        std::string name = parameters.name;
+        if ( !auxiliary::ends_with( name, ".bp" ) )
+        {
+            name += ".bp";
+        }
+
+        auto res_pair = getPossiblyExisting( name );
+        InvalidatableFile shared_name = InvalidatableFile( name );
+        VERIFY_ALWAYS(
+            !( m_handler->accessTypeBackend == AccessType::READ_WRITE &&
+               ( !std::get< PE_NewlyCreated >( res_pair ) ||
+                 auxiliary::file_exists( fullPath(
+                     std::get< PE_InvalidatableFile >( res_pair ) ) ) ) ),
+            "Can only overwrite existing file in CREATE mode." );
+
+        if ( !std::get< PE_NewlyCreated >( res_pair ) )
+        {
+            auto file = std::get< PE_InvalidatableFile >( res_pair );
+            m_dirty.erase( file );
+            dropFileData( file );
+            file.invalidate( );
+        }
+
+        std::string const dir( m_handler->directory );
+        if ( !auxiliary::directory_exists( dir ) )
+        {
+            auto success = auxiliary::create_directories( dir );
+            VERIFY( success, "Could not create directory." );
+        }
+
+        associateWithFile( writable, shared_name );
+        this->m_dirty.emplace( shared_name );
+        getFileData( shared_name ).m_mode = adios2::Mode::Write; // WORKAROUND
+        // ↑ ADIOS2 does not yet implement ReadWrite Mode
```

Comment:
- pls remove non-ascii in code seen by compiler
- work-around: pls open and link corresponding adios2 issue :)